pkg/packageinstall/packageinstall.go

```diff
@@ -201,6 +205,74 @@ func (pi *PackageInstallCR) reconcileAppWithPackage(existingApp *kcv1alpha1.App,
 	return reconcile.Result{}, nil
 }
 
+func (pi *PackageInstallCR) clusterVersionConstraintsSatisfied(pkg *datapkgingv1alpha1.Package) bool {
+	if pkg.Spec.KubernetesVersionSelection == nil || pkg.Spec.KubernetesVersionSelection.Constraints == "" {
+		return true
+	}
+	const kubernetesVersionOverrideAnnotation = "packaging.carvel.dev/ignore-kubernetes-version-selection"
+
+	_, found := pi.model.Annotations[kubernetesVersionOverrideAnnotation]
+	if found {
+		pi.log.Info("Found kubernetes version override annotation; not applying version constraints")
+		return true
+	}
+
+	vi, err := deploy.GetClusterVersion(pi.coreClient, pi.model.Spec.ServiceAccountName, pi.model.Spec.Cluster, &pi.model.ObjectMeta, pi.log)
```

Comment:
clusterVersionConstraintsSatisfied gets called per every package, so we should move out GetClusterVersion outside to avoid N calls.